### DISCUSSION TITLE
Set Time

### DISCUSSION THREAD

**Discussion by heffneil** [aqualinkd/AqualinkD#167] [General]
Hey guys I just - finally - got this working and setup after messing around since last winter!  It appears to operate my pump and waterfall which is good and its pretty speedy.  The one thing I would really like is for the time to sync on the pi with the unit.  I have the correct time on the pi and I had that option enabled in the config.  I am on a PDA only panel so I am not sure if this is possible.  Please let me know.

AuqalinkD: 2.2.2
Aqualink RS PSA-PS6 Combo PPD PDS 5.0

As a feature request, it would be nice to have an Info link where I can copy and paste that data in the future and some other output for the devices might be good for debugging.  I had to type the above and hope I don't mistype or transcribe the digits!

Thanks!

Neil

**Comment by sfeakes** [comment]
Click the AqualinkD in the title ribbon, it’ll display that information.
Time on PDA is hit n miss, it’s a real pain, works fine on some panel versions but highly unreliable on other panel versions.  It’s why AqualinkD always states PDA is in beta.  
what do you see in the logs?, if time is out of sync, you should see AqualinkD trying to set the panel time and log the error.

**Comment by heffneil** [comment]
I got this information from the ribbon, but I couldn't copy and paste it. 

Today I noticed that the web page showed the pump running when it was not.  Not sure whats going on.  Also how can I see and alter programs?

Thanks!

**Comment by sfeakes** [comment]
PDA can be delayed on status updates, but thats dependent on you specific config, usually not more than 30 seconds or so unless you have a device blocking the AqualinkD connection.  Have you looked in the logs for serial connection issues.  It kinda sounds like AqualinkD doesn’t have a clean connection/ dropping packets.

**Comment by heffneil** [comment]
This was wrong for hours after it turned off and I continued to check it.   It even shows it running now when I run my pump between 12am and 8 am.  So its not acting right.  I was able to control things yesterday when I initially got everything running. I must admit I don't know how to check logs.  Are they available in the web interface or where would the logs be located?

Thanks!

**Comment by heffneil** [comment]
I got this:

Apr-14-22 21:19:15 PM Notice:  AqualinkD: Listening to Aqualink RS8 on serial port: /dev/ttyUSB0
Apr-14-22 21:19:15 PM Notice:  AqualinkD: Waiting for Control Panel probe
Apr-14-22 21:19:16 PM Notice:  AqualinkD: Got message but no probe on '0x61', did we start too soon? (waiting for probe)
Apr-14-22 21:19:19 PM Notice:  AqualinkD: Starting communication with Control Panel
Apr-14-22 21:19:22 PM Notice:  Panl Prog: Programming: Init PDA, 10
Apr-14-22 21:19:22 PM Warning: RS Serial: Serial read bad Jandy checksum, ignoring
Apr-14-22 21:19:22 PM Warning: RS Serial: BAD PACKET To 0x61 of type PDA Init (*guess | HEX: 0x10|0x02|0x61|0x1b|0x01|0x00|0x8f|0x10|0xfb|0x10|0x02|0x61|0x1b|0x01|0x00|0x8f|0x10|0x03|
Apr-14-22 21:19:22 PM Error:   AqualinkD: Nothing read on blocking serial port
Apr-14-22 21:19:22 PM Error:   AqualinkD: Aqualink daemon looks like serial error, resetting.
Apr-14-22 21:19:34 PM Notice:  Panl Prog: Programming: PDA init after wake, 15

**Comment by sfeakes** [comment]
That bad packet then nothing read is the issue.  (You probably have that repeated many times)  You have a RS485 connection problem.   So AqualinkD can’t make a stable connection to the control panel.  Check wiring and termination.

**Comment by heffneil** [comment]
This isn't a config issue?  Yesterday I was able to control everything and no one was in the panel after that time?  I am using: FTDI USB-RS485-WE-1800-BT Cable, USB to RS485 Serial, 1.8M, Wire END

**Comment by sfeakes** [comment]
I doubt it’s config issue unless you change some parameters on how the connection works.
The below could potentially effect what you are seeing.  BUT only if you changed them between it working and not working.   It’s far more lightly that a RS485 devise is interfering with the connection. 
‘’’
# Not documented. These will change how RS485 / Serial works, Only use if asked to for problem solving purposes.
#serial_readahead_b4_write = yes
#thread_netservices = yes
#rs_poll_speed = -1
#swg_zero_ignore_count = 20
‘’’

**Comment by heffneil** [comment]
nah I didn't touch these.  I don't know what could have caused the problems.  Frustrating.

**Comment by heffneil** [comment]
is the address potentially a problem? '0x61'

**Comment by heffneil** [comment]
Ok - you were correct the ground came out of the terminal block.  All re-installed and seems to be able to control things ok - I see there's no real way for me to program the panel remotely.

What is with the time sync?  What is that a challenge and anything I can do to make it reliable?  Thats a huge motivator for me to run this given power outages and the time is way off pretty regularly.

**Comment by ballle98** [comment]
There is an open pull request that adds time sync support to PDA systems - https://github.com/sfeakes/AqualinkD/pull/94

**Reply by heffneil** [reply]
Is this not within your version?

**Reply by ballle98** [reply]
Yes this an several other PDA fixes are available here - https://github.com/ballle98/AqualinkD/wiki#install-configuration

**Comment by heffneil** [comment]
I thought I had used your git for the install so I would have the PDA files?  Do the logs not reflect that?  Sorry I thought I followed the correct installation/repository.

**Reply by ballle98** [reply]
I don't change any of the versioning.  If you start aqualinkd this way https://github.com/ballle98/AqualinkD/wiki#run-in-foreground-with-logging  the logs on the console will have fractional time stamps you will also be able to capture the reference to set time.

**Comment by heffneil** [comment]
I am pretty sure I followed those directions and are running your version since you suggested it back on the forums since your version was more appropriate for my PDA panel.

**Comment by ballle98** [comment]
Look/grep for these logs

```
16:28:13.927 Notice:  AqualinkD: Keep panel time in sync  = YES
...
16:28:18.189 Debug:   PDA:       process_pda_packet_msg_long_time strptime Thu Apr 28 16:27:18 2022
....

```
.

**Comment by ballle98** [comment]
The discussion for my version moved here - https://github.com/ballle98/AqualinkD/issues/61

**Comment by heffneil** [comment]
I get to here:

jandy@jandypi:~/git $ git checkout dev/rebase-1.3.x
fatal: not a git repository (or any of the parent directories): .git

Any ideas?

Should I be using the rebase 1.3?

**Comment by heffneil** [comment]
I had to enter the newly made directory to continue. The question is do I want the rebate 1.3.x?

**Reply by ballle98** [reply]
I will delete the instruction for rebase 1.3.x  The 2.2.x versions that are on master are much better now.

Do a "git checkout master" or delete the AqualinkD directory and start over.
